test/odbc/psqlodbc/test_pg_datatype_float.cpp

```diff
@@ -0,0 +1,955 @@
+  
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < inserted_values.size(); ++i) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      ASSERT_EQ(col_results[j], StringToDouble(inserted_values[i][j]));
+      ASSERT_EQ(col_len[j], BYTES_EXPECETED);
+    }
+  }
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  for (int i = 0; i < inserted_values.size(); i++) 
+  {
+    odbcHandler.CloseStmt();
+    vector<pair<string,string>> update_col;
+
+    // setup update column
+    for (int j = 0; j <NUM_COLS; j++) {
+      update_col.push_back(pair<string,string>(COL_NAMES[j], updated_values[i][j]));
+    }
+
+    // Update value and assert an error is present
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(),
+                          (SQLCHAR*) UpdateTableStatement(TABLE_NAME, update_col, COL_NAMES[0] + "='" + PK_VAL + "'").c_str(), 
+                          SQL_NTS);
+    ASSERT_EQ(rcode, SQL_ERROR);
+
+    odbcHandler.CloseStmt();
+
+    odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    // Assert that the results did not change
+    for (int j = 0; j < NUM_COLS; j++) {
+      ASSERT_EQ(col_results[j], StringToDouble(inserted_values[i][j]));
+      ASSERT_EQ(col_len[j], BYTES_EXPECETED);
+    }
+  }
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Float,Arithmetic_Operators) {
+  const int BUFFER_LENGTH=8192;
+  const int BYTES_EXPECTED = 8;
+
+  unsigned char pk;
+  unsigned char data[BUFFER_LENGTH];
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  vector <string> inserted_pk = {
+    "8"
+  };
+
+  vector <string> inserted_data = {
+    "2"
+  };
+
+  vector <string> operations_query = {
+    COL_NAMES[0] + "+" + COL_NAMES[1],
+    COL_NAMES[0] + "-" + COL_NAMES[1],
+    COL_NAMES[0] + "*" + COL_NAMES[1],
+    COL_NAMES[0] + "/" + COL_NAMES[1],
+    "ABS("+COL_NAMES[0]+")",
+    "POWER("+COL_NAMES[0]+","+COL_NAMES[1]+")",
+    "||/ "+COL_NAMES[0],
+    "LOG("+COL_NAMES[0]+")"    
+  };
+
+  vector<vector<string>>expected_results = {{},{}};
+
+  // initialization of expected_results
+  for (int i = 0; i < inserted_pk.size(); i++) {
```

Comment:
is this loop necessary? It looks like it would run once and since the values are hardcoded, it would not match if we added one to inserted_pk.